vllm/config/__init__.py

```diff
@@ -1197,12 +1197,6 @@ def _verify_quantization(self) -> None:
             # Verify quantization configurations.
             if self.quantization is None:
                 self.quantization = quant_method
-            elif self.quantization != quant_method:
-                raise ValueError(
-                    "Quantization method specified in the model config "
-                    f"({quant_method}) does not match the quantization "
-                    f"method specified in the `quantization` argument "
-                    f"({self.quantization}).")
```

Comment:
Surely this error is still relevant? What was the reason for removal?

Reply:
will probably revert this change, was experimenting with on the fly quantization.

we have two path: 1. weight_loader 2. process_weights_after_loading

I'm implementing 1. in the PR, and will add some prototype code for 2. as well.

https://github.com/vllm-project/vllm/pull/23901/files#diff-c1cbb1d99298476e618b6fc564452c8cd9b3ecbf688c5cfb44506209547f7f3d is trying to implement 2, and we'll migrate to that a bit later.

current PR is still WIP, not the final state that we want to land yet

Reply:
this is reverted now, turns out we don't need to make this change